File: RPS.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

void choice(int);

int main()
{
	int user, comp;
	//rock paper scisccor
	//0 1 2

	while(1 < 2)
    {
		printf("1. rock \t2. paper\n3. scicsor\t4. end\n");
        printf("Select anything you want:");
		scanf("%d",&user);

		if(user < 1 || user > 4)
        {
			printf("Enter correct choice\n");
            printf("\n");
			continue;
		}

		if(user == 4)
		break;

		printf("player chose ");
		choice(user);

		comp = (rand()%3)+1;
		printf("computer chose ");
		choice(comp);

		user = pow(2,user-1);
		comp = pow(2,comp-1);

		//process
		if(user == comp)
        {
			printf("draw!, try again\n");
            printf("\n");
			continue;
		}
		else
		 {
			if(user+comp != 5)
            {
				if(user > comp)
				   printf("user won!\n");
				else
				   printf("computer won!\n");
            }
            else
			{
				if(user < comp)
                   printf("user won!\n");
				else
				   printf("computer won!\n");
            }
         }
		printf("\n");
	}
	printf("End.\nSee you later:)");
}

void choice(int n)
 {

	switch(n)
    {
		case 1:
			printf("rock\n");
			break;
		case 2:
			printf("paper\n");
			break;
		case 3:
			printf("scissor\n");
			break;
	}

 }

File: RPS.java
import java.util.*;
import java.lang.Math;

public class RPS
{
    static void choice(int n)
    {
        switch(n)
        {
            case 1:
                System.out.println("rock");
                break;
            case 2:
                System.out.println("paper");
                break;
            case 3:
                System.out.println("scissors");
                break;
        }
    }
    public static void main(String[] args)
    {
        Scanner sc=new Scanner(System.in);
        int user,comp;
        while(true)
            {
                System.out.println("1.rock\t 2.Paper");
                System.out.println("3.Scissors\t4.end");
                System.out.print("Select anything you want: ");
                user=sc.nextInt();
            if(user<1||user>4)
            {
                System.out.println("Enter correct choice");
                continue;
            }
            if(user==4)
            {
                break;
            }
            System.out.print("Player chose ");
            choice(user);

            comp=(int)(Math.random()*3)+1;

            System.out.print("computer chose ");
            choice(comp);

            int u=(int)Math.pow(2,user-1);
            int c=(int)Math.pow(2,comp-1);

            if(u==c)
            {
                System.out.println("draw!, try again\n");
                continue;
            }

            if(u+c!=5)
            {
                if(u>c)
                {
                    System.out.println("user won!");
                }
                else
                    System.out.println("computer won!");
            }
            else
            {
                if(u<c)
                {
                    System.out.println("user won!");
                }
                else
                    System.out.println("computer won!");
            }
            System.out.println();
            
        }
        System.out.println("End");
        System.out.println("See you later:)");
    }
}